design/object-metadata-s3.md

```diff
@@ -0,0 +1,49 @@
+# Object Metadata and S3
+
+### Overview
+
+lakeFS keeps the following metadata per object:
+
+1. Physical address
+1. Last modified time
+1. Size
+1. ETag
+1. Metadata
+1. Address type (relative / full)
+
+### What is missing?
+
+Mapped two issues that we currently do not address when we work with our S3 interface:
+
+1. Content-Type - Not keeping the posted object content-type and do not set it back when we get the object.
+  [Support Content-Type](https://github.com/treeverse/lakeFS/issues/2296)
+  We saw this issue when one of the clients used the value of the content type to identify a directory marker.
+  Assuming there are more use-cases where the content-type is set and/or used.
```

Comment:
More generally, because this exists in S3, it is reasonable for a user of the S3 gateway to expect that it will work.